mapflow/functional/layer_utils.py

```diff
@@ -361,15 +362,54 @@ def download_results_file(self, pid) -> None:
             timeout=300
         )
 
+    def download_aoi_file(self, pid) -> None:
+        """
+        Download area of interest and save to a geojson file
+        """ 
+        path = Path(self.temp_dir)/f'{pid}_aoi.geojson'                         
+        self.dlg.toolButton_2.setEnabled(False)
+        self.http.get(
+            url=f'{self.server}/processings/{pid}/aois',
+            callback=self.download_aoi_file_callback,
+            callback_kwargs={'path': path},
+            use_default_error_handler=True,
+            timeout=30
+        )
+
     def download_results_file_callback(self, response: QNetworkReply, path: str) -> None:
         """
         Write results to the geojson file
         """
-        self.dlg.saveResultsButton.setEnabled(True)
+        self.dlg.toolButton_2.setEnabled(True)
         with open(path, mode='wb') as f:
             f.write(response.readAll().data())
         self.message_bar.pushSuccess(self.tr("Results saved"),
                                      self.tr("see in {path}!").format(path=path))
+        
+    def download_aoi_file_callback(self, response: QNetworkReply, path: str) -> None:
+        """
+        Write area of interest to the geojson file
+        """
+        self.dlg.toolButton_2.setEnabled(True)
+        data = json.loads(response.readAll().data())
+        geojson = {
+        "type": "FeatureCollection",
+        "features": [{
+            "type": "Feature",
+            "geometry" : d['geometry'],
+            "properties" : {"id": d['id'], 
+                            "status": d['status'],
+                            "percentCompleted": d['percentCompleted'],
+                            "area": d['area'],
+                            "messages": d['messages']},
+            }for d in data]
+        }
+        with open(path, "w") as f:
+            json.dump(geojson, f)
+        id = os.path.splitext(os.path.basename(path))[0][:-4]
```

Comment:
This looks not very good - parsing the string here.

I suggest two options:
1. Pass PID, not Path, and construct path here
2. use Pathlib to get the name (it is `path.stem`) from Path object